### ISSUE TITLE
Empty table header found in the 'Your locked pages' dashboard section

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#11459]
### Issue Summary

Initially reported by @albinazs as part of [WCAG 2.2 AAA\* audit of the Wagtail admin – Nov 2023](https://github.com/wagtail/wagtail/discussions/11180).

We seem to be using an empty table header and table cells for spacing, to align this report with others on the page. That’s fine, but the column should then be hidden from screen reader users so they don’t hear “blank” when navigating the table (it’s confusing and makes it harder to navigate the table). This is likely a failure of `1.3.1 Info and Relationships`.

### Steps to Reproduce

See demo: [Your locked pages section](https://static-wagtail-v5-2.netlify.app/admin/#locked-pages-section)

See https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templates/wagtailadmin/home/locked_pages.html#L59, and the corresponding `<th></th>`.

### Working on this

Anyone can contribute to this. This should be a simple template change. Make sure you know how to test this with a screen reader. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by arnavchhokra** [comment]
@thibaudcolas 
hi, can i work on this?

**Comment by GetRohitansh** [comment]
I guess without changing much, just adding "aria-hidden=true" attribute to <th> might work, please tell @thibaudcolas or should I approach some other method.

**Comment by lb-** [comment]
@GetRohitansh give that a go, load up a screen reader and check, sounds like a good idea.
Another approach can be `role="presentation"`
see https://www.a11ytips.dev/docs/roles-vs-presentation/

**Comment by lb-** [comment]
@GetRohitansh  as per comment on #11465 

There are another bunch of `<th></th>` in Wagtail, some in templates and some in other generated code.

Feel free to raise a new PR that cleans these up also, remember to create a new branch for this work and read the contributors guide first though :) 

As for the `<th></th>`, it appears that some are in JavaScript, if updating these be sure to update the Jest snapshots (`npm run test:unit -- -u`) but also ensure that the JS code does not 'fill these in' with titles somehow first.

Here's the files

1. https://github.com/wagtail/wagtail/blob/c99c74cc7e5f7d534fecb6ab0be5a9c7c4da682a/client/src/entrypoints/contrib/typed_table_block/typed_table_block.js#L79
2. https://github.com/wagtail/wagtail/blob/c99c74cc7e5f7d534fecb6ab0be5a9c7c4da682a/wagtail/admin/templates/wagtailadmin/permissions/includes/collection_member_permissions_formset.html#L29
3. https://github.com/wagtail/wagtail/blob/c99c74cc7e5f7d534fecb6ab0be5a9c7c4da682a/wagtail/admin/templates/wagtailadmin/workflows/includes/workflow_pages_formset.html#L20
4. https://github.com/wagtail/wagtail/blob/c99c74cc7e5f7d534fecb6ab0be5a9c7c4da682a/wagtail/admin/ui/tables/pages.py#L75
5. https://github.com/wagtail/wagtail/blob/c99c74cc7e5f7d534fecb6ab0be5a9c7c4da682a/wagtail/users/templates/wagtailusers/groups/includes/formatted_permissions.html#L123
6. https://github.com/wagtail/wagtail/blob/c99c74cc7e5f7d534fecb6ab0be5a9c7c4da682a/wagtail/users/templates/wagtailusers/groups/includes/page_permissions_formset.html#L28


wagtail/admin/static_src/wagtailadmin/js/vendor/jquery.datetimepicker.js (ignore this file, it's going to be removed in the near future)
